cloudinit/tests/test_conftest.py

```diff
@@ -16,3 +18,23 @@ def test_typeerrors_on_incorrect_usage(self):
     @pytest.mark.parametrize('disable_subp_usage', [False], indirect=True)
     def test_subp_usage_can_be_reenabled(self):
         util.subp(['whoami'])
+
+
+class TestDisableSubpUsageInTestSubclass(CiTestCase):
+    """Test that disable_subp_usage doesn't impact CiTestCase's subp logic."""
+
+    def test_using_subp_raises_assertion_error(self):
+        with pytest.raises(Exception):
```

Comment:
You should fix the comment 'assertion_error'.  Or fix the _fake_subp to *raise* an assertion_error . Otherwise this is confusing.  Your test is named `....raises_assertion_error...` and then it insists that the more generic Exception is raised.

Reply:
Yep, this was a copy/paste error from the above tests, will fix.

(The pytest functionality raises `AssertionError` but I didn't want to change the behaviour of `CiTestCase` "just" for the sake of consistency.)